src/restconfClient.ts

```diff
@@ -46,19 +46,39 @@ export class RestconfClient {
         return clients.filter((c: any) => c.id !== this.environment.migrationClientId);
     }
 
+    public async deleteClient(profileId: string, clientId: string): Promise<void> {
+
+        console.log(clientId);
+        const path = `base:profiles/profile=${profileId},oauth-service/settings/profile-oauth:authorization-server/client-store/config-backed/client=${clientId}`;
+        await this.deleteData(path);
+    }
+
     private async getData(path: string): Promise<any> {
 
         const response = await fetch(`${this.restConfApiBaseUrl}/${path}`, {
+            method: 'GET',
             headers: this.getRequestHeaders(),
         });
 
         if (response.status !== 200) {
-            throw new Error(`RESTCONF request to ${path} failed: ${response.statusText}`);
+            throw new Error(`RESTCONF GET request to ${path} failed: ${response.statusText}`);
         }
 
         return await response.json();
     }
 
+    private async deleteData(path: string): Promise<any> {
+
+        const response = await fetch(`${this.restConfApiBaseUrl}/${path}`, {
+            method: 'DELETE',
+            headers: this.getRequestHeaders(),
+        });
+
+        if (response.status !== 204) {
+            throw new Error(`RESTCONF DELETE request to ${path} failed: ${response.statusText}`);
```

Comment:
It seems that you could go almost to the end of the migration, then finally fail here and stop everything.
I would suggest trying to do a migration fully before attempting any deletions, and then do deletions as a single transaction which either succeeds or fails atomically.

Reply:
That's a good point: I changed the code to output a warning if deletion fails.
Status codes 204 and 404 are expected, so they do not generate the warning.